docs/src/main/asciidoc/config-extending-support.adoc

```diff
@@ -0,0 +1,338 @@
+`Iterable<ConfigSource> getConfigSources(ConfigSourceContext context)` method. The `ConfigSourceFactory` may also
+assign a priority by overriding the default method `OptionalInt getPriority()`. This is only used to sort multiple
+factories during initialization. Once all the factories are initialized, the provided list of `ConfigSource` will use
+their own ordinal and sorted with all the other sources available in the `Config` instance.
+
+When the Factory is initializing, the provided `ConfigSourceContext` may call the method
+`ConfigValue getValue(String name)`. This method lookups configuration names in all `ConfigSource` that were already
+initialized by the `Config` instance, including sources with lower ordinals than the ones defined in the
+`ConfigSourceFactory`. The `ConfigSource` list provided by a `ConfigSourceFactory` is not taken into considerations to
+configure other sources produced by a lower priority `ConfigSourceFactory`.
+
+[[custom-converters]]
+== Custom Converters
+
+It is possible to create a Custom Converter type as specified by
+link:https://github.com/eclipse/microprofile-config/blob/master/spec/src/main/asciidoc/converters.asciidoc#adding-custom-converters[MicroProfile Config Custom Converters].
+
+A custom Converter requires an implementatin of `org.eclipse.microprofile.config.spi.Converter<T>`. Each implementation
+requires registration via the https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html[ServiceLoader]
+mechanism, either in `META-INF/services/org.eclipse.microprofile.config.spi.Converter`. Consider:
+
+[source,java]
+----
+package org.acme.config;
+
+public class MicroProfileCustomValue {
+
+    private final int number;
+
+    public MicroProfileCustomValue(int number) {
+        this.number = number;
+    }
+
+    public int getNumber() {
+        return number;
+    }
+}
+----
+
+The corresponding converter will look like the one below. Please note that the custom converter class must be `public`
+and must have a `public` no-argument constructor. It also must not be `abstract`.
+
+[source,java]
+----
+package org.acme.config;
+
+import org.eclipse.microprofile.config.spi.Converter;
+
+public class MicroProfileCustomValueConverter implements Converter<MicroProfileCustomValue> {
+
+    @Override
+    public MicroProfileCustomValue convert(String value) {
+        return new MicroProfileCustomValue(Integer.parseInt(value));
+    }
+}
+----
+
+The custom configuration type converts the Config Source value automatically:
+
+[source,java]
+----
+@ConfigProperty(name = "configuration.value.name")
+MicroProfileCustomValue value;
+----
+
+=== Converter priority
+
+The `javax.annotation.Priority` annotation overrides the Converter priority and change converters precedence to fine
+tune converters execution order. By default, if no `@Priority` can be found on a converter, it's registered with a
+priority of `100` and all Quarkus core converters are registered with a priority of `200`. Consider:
```

Comment:
I think we should say explicitly that to override a Quarkus converter, you need a 200+ priority (at least, that's what I understood).

Reply:
Added as note.